File: 5/list-5.go
package main

import "fmt"

type Animal interface {
	Som() string
}

type Cachorro struct{}

func (c Cachorro) Som() string {
	return "Au au"
}

type Gato struct{}

func (g Gato) Som() string {
	return "Miau"
}

func fazerSomDeAnimais(animais []Animal) {
	for _, animal := range animais {
		fmt.Println(animal.Som())
	}
}

func main() {
	animais := []Animal{
		Cachorro{},
		Gato{},
		Cachorro{},
	}

	fazerSomDeAnimais(animais)
}

File: 5/list-5.py
class Animal:
    def som(self) -> str:
        return "som que animal faz"

class Cachorro(Animal):
    def som(self) -> str:
        return "Au au"

class Gato(Animal):
    def som(self) -> str:
        return "Miau"

def fazer_som(animais):
    for animal in animais:
        print(animal.som())

def main():
    animais = [Cachorro(), Gato()]
    fazer_som(animais)

if __name__ == "__main__":
    main()
